### PR TITLE
feat: c-api

### PR DESCRIPTION
**Pull request by muhmud**

### CODE REVIEW

File: examples/skia-demo-player-c/main.cpp [hunk 1]
```diff
@@ -0,0 +1,178 @@
+#include <X11/Xlib.h>
```
Comment (hermet) [inline examples/skia-demo-player-c/main.cpp:1]: Relying on the X Window System and Skia is a bad choice for portability. How about using SDL2 instead?

**Review by theashraf — COMMENTED** [review]
Can we add a README.md file to the C examples to explain how to run them, including the requirements for different platforms (Mac/Linux/Windows)?

It would also be great to update the BUILD_SYSTEM.md with the recent changes and the README.md with an introduction to the new C bindings and instructions on how to use them.

File: dotlottie-ffi/src/dotlottie_player.cpp
[the remark this answers is not in the document]
Comment (muhmud) [reply dotlottie-ffi/src/dotlottie_player.cpp:1]: Yep, removed

File: dotlottie-ffi/test/a.out
[the remark this answers is not in the document]
Comment (muhmud) [reply dotlottie-ffi/test/a.out:1]: Removed

File: examples/demo-player-c/None
[the remark this answers is not in the document]
Comment (muhmud) [reply examples/demo-player-c/None:1]: Removed

File: dotlottie-ffi/src/dotlottie_player.hpp
[the remark this answers is not in the document]
Comment (muhmud) [reply dotlottie-ffi/src/dotlottie_player.hpp:1]: Removed

File: examples/skia-demo-player-c/main.cpp (hunk 1 above)
Comment (muhmud) [reply examples/skia-demo-player-c/main.cpp]: Updated the examples to use SDL2

**Comment by muhmud** [comment]
Added/Updated `README` files, as requested, and also `BUILD_SYSTEM.md`

**Review by theashraf — APPROVED** [review]

**Pull request merged**